.github/workflows/test-and-publish-on-push-to-main.yml

```diff
@@ -0,0 +1,96 @@
+name: Test (and Publish to Docker Hub on push to main)
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+    - main
+    - github-actions
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        flavor:
+          - octodns
+          - azure
+          - cloudflare
+          - constellix
+          - digitalocean
+          - dnsimple
+          - dnsmadeeasy
+          - dyn
+          - easydns
+          - edgedns
+          - etchosts
+          - gandi
+          - gcore
+          - googlecloud
+          - hetzner
+          - mythicbeasts
+          - ns1
+          - ovh
+          - powerdns
+          - rackspace
+          - route53
+          - selectel
+          - transip
+          - ultra
+    steps:
+      - uses: actions/checkout@v3
+      - name: set up qemu virt with platforms
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: arm64,arm
+      - name: set up docker buildx builder with platforms
+        uses: docker/setup-buildx-action@v2
+        with:
+          platforms: linux/amd64,linux/arm64,linux/arm/v7
+      - name: login to the Docker registry
+        uses: docker/login-action@v2
+        with:
+          registry: docker.io
+          username: octodns
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: create metadata (tags and labels)
+        uses: docker/metadata-action@v4
+        id: meta
+        with:
+          images: octodns/${{ matrix.flavor }}
+          flavor: |
+            latest=false
+          tags: |
+            type=raw,value={{date 'YYYY.MM'}}
+          labels: |
+            org.opencontainers.image.documentation=https://github.com/octodns/octodns#readme
+            org.opencontainers.image.licenses=MIT
+      - name: build test image
+        uses: docker/build-push-action@v4
+        with:
+          context: all
+          load: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          build-args: OCTODNS_FLAVOR=${{ matrix.flavor }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+      - name: test image has executables on PATH
+        run: |
+          docker run --rm ${{ steps.meta.outputs.tags }} which octodns-compare octodns-dump octodns-report octodns-sync octodns-validate
+      - name: test image executable will run
+        run: |
+          docker run --rm ${{ steps.meta.outputs.tags }} octodns-sync --help
+      - name: push to registry (unless pull request)
+        uses: docker/build-push-action@v4
+        with:
+          context: all
+          push: ${{ github.event_name != 'pull_request' }}
```

Comment:
This may push for any other event, like tagging. Could we filter by event of push with target ref of main?